tests/foreman/cli/test_contentview.py

```diff
@@ -3126,6 +3127,87 @@ def test_positive_inc_update_no_lce(self, module_org, module_product, module_tar
+        """A CCV containing a CV that has an incremental update applied should auto-publish
+
+        :Verifies: SATQE-17141
+
+        :id: b36e3fd1-7dfe-4c71-aaed-deafb10f577e
+
+        :customerscenario: true
+
+        :expectedresults: CCV is auto-published when composite CV recieves an inc update
+
+        :CaseImportance: Medium
+
+        """
+        repo = module_target_sat.cli_factory.make_repository(
+            {
+                'product-id': module_product.id,
+                'content-type': 'yum',
+                'url': settings.repos.yum_1.url,
+            }
+        )
+        module_target_sat.cli.Repository.synchronize({'id': repo['id']})
+        content_view = module_target_sat.cli_factory.make_content_view(
+            {'organization-id': module_org.id, 'repository-ids': repo['id']}
+        )
+        module_target_sat.cli.ContentView.add_repository(
+            {
+                'id': content_view['id'],
+                'organization-id': module_org.id,
+                'repository-id': repo['id'],
+            }
+        )
+        cvf = module_target_sat.cli_factory.make_content_view_filter(
+            {
+                'content-view-id': content_view['id'],
+                'inclusion': 'true',
+                'name': gen_string('alpha'),
+                'type': 'rpm',
+            },
+        )
+        module_target_sat.cli_factory.content_view_filter_rule(
+            {
+                'content-view-filter-id': cvf['filter-id'],
+                'name': FAKE_2_CUSTOM_PACKAGE_NAME,
+                'version': 5.21,
+            }
+        )
+        module_target_sat.cli.ContentView.publish({'id': content_view['id']})
+        content_view = module_target_sat.cli.ContentView.info({'id': content_view['id']})
+        assert len(content_view['versions']) == 1
+        cv_version = content_view['versions'][0]
+        composite_view = module_target_sat.cli_factory.make_content_view(
+            {'composite': True, 'auto-publish': 'true', 'organization-id': module_org.id}
+        )
+        module_target_sat.cli.ContentView.component_add(
+            {
+                'composite-content-view-id': composite_view['id'],
+                'component-content-view-id': content_view['id'],
+                'latest': True,
+            }
+        )
+        # Check if Composive CV has been published before and after incremental update
+        composite_view = module_target_sat.cli.ContentView.info({'id': composite_view['id']})
+        assert len(composite_view['versions']) == 0
+        module_target_sat.cli.ContentView.version_incremental_update(
+            {
+                'content-view-version-id': cv_version['id'],
+                'errata-ids': settings.repos.yum_1.errata[1],
+            }
+        )
+        # Gives time for the incremental update to finish, and propagate a publish of the composive cv
+        time.sleep(2)
```

Comment:
I can see "Publish CV" task being spawned after incremental update. Should we wait for them to finish instead of the sleep?
```suggestion
            module_target_sat.wait_for_tasks(
            search_query=(
                f'Actions::Katello::ContentView::Publish and organization_id = {module_org.id}'
            ),
            max_tries=5,
            poll_rate=10,
        )
```

Reply:
This is definitely a more deterministic way to do this. I'll give it a try. Should be consistent.